Fixes for remaining issues with boolean parameters

katzyn opened the pull request:
I found more issues with boolean parameters. For example, `IFEXISTS=1` in connection URL worked as `IFEXISTS=FALSE`, this parameter also use one more code path for parsing, there are too many of them.

1. I missed that there is also `ConnectionInfo.removeProperty()` for boolean values, not only `getProperty()`. `removeProperty()` was changed to use `Utils.parseBoolean()` too.

2. WebApp and CSV also have similar issues, so I used `Utils.parseBoolean()` in them.

3. SessionRemote has read booleans as strings from ConnectionInfo with following calls to `Boolean.parseBoolean()`, I changed it to use specialized variant of `getProperty()` for boolean values that is already fixed in master.

4. Conversion from `ValueString` to `ValueBoolean` was more advaced than `Utils.parseBoolean()` so I implemened support for additional values in `Utils.parseBoolean()` and replaced code in conversion method with call to `Utils.parseBoolean()` so we have only one place with custom parsing of booleans.

There are some remaining calls to `Boolean.parseBoolean()` in main H2 code. But they seems to use strings from internal data only where custom values are not expected.

katzyn committed:
Use Utils.parseBoolean() in more places with user-supplied strings

katzyn committed:
Use ConnectionInfo.getProperty(String, boolean) in SessionRemote

katzyn committed:
Add support for more strings to Utils.parseBoolean() and use it in more places

grandinj commented:
The first two commits I'm fine with, but not 61a0fcf, that looks dodgy. Lets leave that extra logic in Value alone for now.
In particular, that
    new BigDecimal(s).signum() != 0
looks like a step backwards for the utility code

katzyn commented:
If I reimplement last `!= 0` check without `BigDecimal` with simple string scan it will be suitable for this utility method?

katzyn commented:
Or we don't want to support 'y', 'n', 'yes', 'no', etc in this method at all?

grandinj commented:
Leave out BigDecimal and I'm fine with the other changes

katzyn commented:
One more question to be sure. Reimplement check for 0.000 etc without BigDecimal, or remove this check at all from `Utils` and revert `Value.convertTo()` to old custom parsing of all these 'y', 't', 'no', etc?

grandinj commented:
I don't like parsing 0.000 as true in the utility code, seems like it's more likely to hide errors than be useful.

But we can't remove it from Value because of compatibility. So rather just leave the Value code alone.

katzyn committed:
Partially revert the previous commit

katzyn commented:
`0.000` was parsed as `false` actually, like `Value` does. OK, I reverted changes in `Value` and removed `BigDecimal` part from `Utils.parseBoolean()` and from tests. So only `0` and `1` numbers parsed as booleans in this utility method. Strings like `0.000` will throw an exception or default value will be used depending on the last parameter.
